src/Microsoft.AspNet.Razor/CodeGenerators/TagHelperAttributeValueCodeRenderer.cs

```diff
@@ -56,6 +56,12 @@ public class TagHelperAttributeValueCodeRenderer
                 throw new ArgumentNullException(nameof(renderAttributeValue));
             }
 
+            if (attributeDescriptor.IsEnum && !complexValue)
```

Comment:
This class is used three times in `CSharpTagHelperCodeRenderer` but the new code is relevant only for `CSharpTagHelperCodeRenderer.RenderRawAttributeValue()`. Suggest moving this addition there.